tmuxp/testsuite/workspacebuilder.py

```diff
@@ -253,6 +253,32 @@ def test_window_options(self):
             window_count += 1
             w.select_layout(wconf['layout'])
 
+    def test_window_shell(self):
+        yaml_config = """
+        session_name: test window options
+        start_directory: '~'
+        windows:
+        - layout: main-horizontal
+          options:
+            main-pane-height: 5
+          panes:
+          - pane
+          - pane
+          - pane
+          window_name: editor
+          window_shell: test_command
+        """
+        s = self.session
+        sconfig = kaptan.Kaptan(handler='yaml')
+        sconfig = sconfig.import_config(yaml_config).get()
+        sconfig = config.expand(sconfig)
+
+        builder = WorkspaceBuilder(sconf=sconfig)
+
+        for w, wconf in builder.iter_create_windows(s):
+            if 'window_shell' in wconf:
+                self.assertEqual(wconf['window_shell'], text_type('test_command'))
+
```

Comment:
These are intended to be integration tests in the truest sense of the word. See ``.travis.yml``.

We have a minimum tmux version we officially support (though we can bump it of course). I'd like to be sure that this behavior would work on older versions.

Try something like this:

```diff
diff --git a/tmuxp/testsuite/workspacebuilder.py b/tmuxp/testsuite/workspacebuilder.py
index 253834e..a8b79dc 100644
--- a/tmuxp/testsuite/workspacebuilder.py
+++ b/tmuxp/testsuite/workspacebuilder.py
@@ -266,7 +266,7 @@ class WindowOptions(TmuxTestCase):
           - pane
           - pane
           window_name: editor
-          window_shell: test_command
+          window_shell: top
         """
         s = self.session
         sconfig = kaptan.Kaptan(handler='yaml')
@@ -277,7 +277,14 @@ class WindowOptions(TmuxTestCase):
 
         for w, wconf in builder.iter_create_windows(s):
             if 'window_shell' in wconf:
-                self.assertEqual(wconf['window_shell'], text_type('test_command'))
+                self.assertEqual(wconf['window_shell'], text_type('top'))
+            for i in range(10):
+                self.session.server._update_windows()
+                if w['window_name'] != 'top':
+                    break
+                time.sleep(.2)
+
+            self.assertNotEqual(w.get('window_name'), text_type('top'))
 
 
 class EnvironmentVariables(TmuxTestCase):
@@ -305,7 +312,7 @@ class EnvironmentVariables(TmuxTestCase):
 
         self.assertEqual('BAR', self.session.show_environment('FOO'))
         self.assertEqual('/tmp', self.session.show_environment('PATH'))
-        
+
 class WindowAutomaticRename(TmuxTestCase):
 
     yaml_config = """
```